src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -589,106 +587,100 @@
-              (.trackColumnForAutoSizing ^SXSSFSheet sheet i))
-            (setup-header-row! sheet (count ordered-cols))
-            (spreadsheet/add-row! sheet (common/column-titles ordered-cols col-settings true)))))
+          (let [{:keys [workbook sheet]} @workbook-data
+                data-format              (. ^SXSSFWorkbook workbook createDataFormat)]
+            (set-no-style-custom-helper sheet)
+            (vreset! cell-styles (compute-column-cell-styles workbook data-format viz-settings ordered-cols))
+            (vreset! typed-cell-styles (compute-typed-cell-styles workbook data-format)))))
 
       (write-row! [_ row row-num ordered-cols {:keys [output-order] :as viz-settings}]
-        (let [ordered-row             (if output-order
-                                        (let [row-v (into [] row)]
-                                          (for [i output-order] (row-v i)))
-                                        row)
-              col-settings            (::mb.viz/column-settings viz-settings)
-              {:keys [pivot-options]} @pivot-data!]
-          (if pivot-options
-            (let [{:keys [pivot-grouping-key]} pivot-options
-                  group                        (get row pivot-grouping-key)]
-              (when (= 0 group)
-                ;; TODO: right now, the way I'm building up the native pivot,
-                ;; I end up using the docjure set-cell! (since I create a whole sheet with all the rows at once)
-                ;; I'll want to change that so I can use the set-cell! method we have in this ns, but for now just string everything.
-                (let [modified-row (->> (vec (m/remove-nth pivot-grouping-key row))
-                                        (mapv (fn [value]
-                                                (if (number? value)
-                                                  value
-                                                  (str value)))))]
-                  (swap! pivot-data! update :rows conj modified-row))))
-            (do
-              (add-row! sheet ordered-row ordered-cols col-settings @cell-styles @typed-cell-styles)
-              (when (= (inc row-num) *auto-sizing-threshold*)
-                (autosize-columns! sheet))))))
+        (let [ordered-row        (if output-order
+                                   (let [row-v (into [] row)]
+                                     (for [i output-order] (row-v i)))
+                                   row)
+              col-settings       (::mb.viz/column-settings viz-settings)
+              pivot-grouping-key @pivot-grouping-key
+              group              (get row pivot-grouping-key)
+              modified-row       (if pivot-grouping-key
+                                   (vec (m/remove-nth pivot-grouping-key ordered-row))
+                                   ordered-row)
+              {:keys [sheet]}    @workbook-data]
+          (when (or (not group)
+                    (= group 0))
```

Comment:
We also only want to keep pivot-grouping = 0 rows. These are the actual data rows we care about, and they're the rows you'll see if you change the viz from a pivot to a regular table. pivot-grouping > 0 are subtotals, and would mess up the pivot table's values (they'd be wrong) if we write them to the file, so we just don't write them